### ISSUE TITLE
Undesired removal points of static object

### ISSUE THREAD

**Issue by youliangtan** [xlab905/velodyne_slam#2]
Hi, Thanks for the code. One issue that I faced was the removal of static point cloud after some time. This created a less dense point cloud in some region with voids. Hope you guide me on which changes to made. Screenshots are shown below

![void in wall](https://user-images.githubusercontent.com/32189404/47332329-855fbe80-d6b1-11e8-994a-c2f449eabed3.png) Voids on wall

![no void in wall](https://user-images.githubusercontent.com/32189404/47332337-8b559f80-d6b1-11e8-9bc2-59a774f9a29b.png) Loam without motion removal

Also, I wanna check with you on the definition of each parameters here: gradThreshold, phiRegion, phiDiffFactor, reserveDistance. Hope you can explain it here 谢了!

**Comment by klai-hss** [comment]
Hi, thanks for pointing out this problem. This code has only been tested on our self-recorded bag files and the parameters are specific to our datasets. Although it will remove some of the static points, but it is not very obvious in our datasets (as shown in the figures). In our method, we transformed the point cloud to the polar coordinate, but this is only a prototype that needs further improvements. The definitions of the mentioned parameters are described below. The phi angle represent the vertical field of view of each LiDAR scan and "phiRegion" is the desired discrepancy between LiDAR scans. This value ranges from 0 to 1. "phiDiffFactor" is controlling how much we want to emphasize on the phi angle. We added "reserveDistance" to see if the position of a point is beyond a certain distance and remove it afterwards. As for "gradThreshold", we used it to do data smoothing. It determines the distance threshold for a point to be removed. In summary, this method should remove moving points that are near the vehicle.
[Before]
![after_simple](https://user-images.githubusercontent.com/28901297/47506653-1cd83380-d8a3-11e8-9c32-b8343fee8f95.png)
[After]
![after](https://user-images.githubusercontent.com/28901297/47506570-ef8b8580-d8a2-11e8-97ae-7e6c67ab4fd2.png)
